tests/unit/plugins/test_writer.py

```diff
@@ -239,6 +250,10 @@ def test_pipeline_generation(
     file_names = [pipeline_name + file_suffix for file_suffix in file_endings]
     if framework == "DVC":
         file_names.append("dvc.yaml")
+        if pipeline_name == "dvc_pipeline_a_b0_stageperartifact":
+            file_names = file_names + [
+                "task_" + art + ".py" for art in artifact_list
```

Comment:
Can you add a TODO line here for testing DVC style frameworks?

The coverage here isn't complete because not all tasks map to an artifact_list. 

Something like the following will work
#TODO fix coverage for tests of file per task frameworks to include non artifact tasks

Not expecting this fix in this PR, getting the task breakdown requires interface breaking look into ArtifactCollection and other internals.

Reply:
Ok